### PR TITLE
[15.0][MIG] stock_archive_constraint: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT36549

### CODE REVIEW

**Comment by cesar-tecnativa** [comment]
@victoralmau please review

**Comment by cesar-tecnativa** [comment]
@pedrobaeza please review

**Comment by cesar-tecnativa** [comment]
@sergio-teruel please review

**Review by pedrobaeza — COMMENTED** [review]
Can you please adjust at least Spanihs translations?

File: stock_archive_constraint/models/product_product.py [hunk 1]
```diff
@@ -21,9 +21,10 @@ def _check_active_stock_archive_constraint_stock_quant(self):
         if res:
             raise ValidationError(
                 _(
-                    "It is not possible to archive product '%s' which has "
-                    "associated stock quantities." % res[0].product_id.display_name
+                    "It is not possible to archive product '%(display_name)s' which has "
+                    "associated stock quantities."
                 )
+                % {"display_name": res[0].product_id.display_name}
```
Comment (rousseldenis) [inline stock_archive_constraint/models/product_product.py]: You can use recordset directly as there is a limit=1.

`res.product_id.display_name`

Same for other lines

**Review by rousseldenis — COMMENTED** [review]
Minor change

**Comment by cesar-tecnativa** [comment]
> Can you please adjust at least Spanihs translations?

done

File: stock_archive_constraint/readme/CONTRIBUTORS.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+* `Tecnativa <https://www.tecnativa.com>`_:
+
+  * Carlos Daudén
+  * Víctor Martínez
+  * César A. Sánchez
```
Comment (victoralmau) [inline stock_archive_constraint/readme/CONTRIBUTORS.rst]:
```suggestion
    * Carlos Daudén
    * Víctor Martínez
    * César A. Sánchez
```

File: stock_archive_constraint/models/stock_location.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            ],
+            limit=1,
+        )
+        if res:
+            raise ValidationError(
+                _(
+                    "It is not possible to archive location '%(display_name)s' which has "
+                    "associated stock quantities."
+                )
+                % {"display_name": res[0].display_name}
```
Comment (victoralmau) [inline stock_archive_constraint/models/stock_location.py]:
```suggestion
                % {"display_name": res.display_name}
```

File: stock_archive_constraint/models/stock_location.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            ],
+            limit=1,
+        )
+        if res:
+            raise ValidationError(
+                _(
+                    "It is not possible to archive location '%(display_name)s' which has "
+                    "associated picking lines."
+                )
+                % {"display_name": res[0].display_name}
```
Comment (victoralmau) [inline stock_archive_constraint/models/stock_location.py]:
```suggestion
                % {"display_name": res.display_name}
```

File: stock_archive_constraint/models/stock_location.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            ],
+            limit=1,
+        )
+        if res:
+            raise ValidationError(
+                _(
+                    "It is not possible to archive location '%(display_name)s' which has "
+                    "associated stock reservations."
+                )
+                % {"display_name": res[0].display_name}
```
Comment (victoralmau) [inline stock_archive_constraint/models/stock_location.py]:
```suggestion
                % {"display_name": res.display_name}
```

**Review by victoralmau — CHANGES_REQUESTED** [review]
Code review OK, some little changes.

Could you squash administrative commits?

**Comment by sergio-teruel** [comment]
@victoralmau Changes done!!

**Review by victoralmau — APPROVED** [review]
Code review OK

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**